File: computer3/Fall Term/Solutions/Class09_Nov_06/lvl3_prime_time.py
import sys
import math
input = sys.stdin.readline

for _ in range(5):
    n = int(input())
    f = [0] * (n+1)
    for i in range(2, n+1):
        x = i
        for j in range(2, int(math.sqrt(x))+1):
            while x % j == 0:
                f[j] += 1
                x //= j
        if x != 1:
            f[x] += 1
    first = True
    for i in range(2, n+1):
        if f[i] > 0 :
            if first:
                print(i, '^', f[i], ' ', sep='', end='')
                first = False
            else:
                print('* ', i, '^', f[i], ' ', sep='', end='')
    print()

File: computer3/Fall Term/Solutions/Class09_Nov_06/lvl3_prime_time.cpp
#include <bits/stdc++.h>
using namespace std;
int f[10002], n, m;
int main(){
    //freopen("test.txt", "r", stdin);
    for(int t=1; t<=5; t++){
        scanf("%d", &m); memset(f, 0, sizeof(f));
        for(int j=2; j<=m; j++){
            int n = j;
            for(int i=2, up=sqrt(n); i<=up; i++){
                while( n % i == 0 ){
                    f[i]++; n/=i;
                }
            }
            if(n != 1) f[n]++;
        }
        bool first = true;
        for(int i=2; i<=m; i++){
            if(f[i] != 0){
                if(!first) printf(" * %d^%d", i, f[i]);
                else { printf("%d^%d", i, f[i]); first = false;}
            }
        }
        printf("\n");
    }
}